### ISSUE TITLE
second user cursor deletes text, prevents newline and inserts username as text

### ISSUE THREAD

**Issue by kolAflash** [nextcloud/text#5103]
**Describe the bug**
@juliushaertl This is a follow up for regressions since >= 26.0, already described in #5102 and #4100

Since Nextcloud 26.0, there is strange behaviour if a user cursor is present.

&nbsp;

**To Reproduce**
A webbrowser (Firefox-120.0.1 on Debian-12) is having the file opened simultaneously. But I'm doing nothing on the webbrowser client. It just has the file opened.

Please see the attached videos for behaviour and reproduction. (recorded on Android)  
Mind the Return and Backspace keys flashing when I press them.

<details>
<summary>VIDEOS</summary>
https://github.com/nextcloud/text/assets/3355089/310995e5-b0eb-4efa-bceb-10b47388fc3b

https://github.com/nextcloud/text/assets/3355089/70b146fe-9525-4245-b480-21fbeafaca0a
</details>

&nbsp;

**Expected behavior**
Text should not be deleted on pressing Return. (related #1927)  
Adding a newline (Return) at a user cursor position should work.  
Username should not be added as text when pressing backspace at a user cursor position.

_(slightly related #5102)_

&nbsp;

**Server details:**
Tested with two different servers.
- server 1
  - Nextcloud version: 26.0.9
  - PHP Version: 8.2 (Debian-12)
  - Database: MariaDB 10.11.4 (Debian-12)
- server 2
  - Nextcloud version: 27.1.1.0
  - PHP Version: unknown
  - Database: unkonwn

&nbsp;

**Client details:**
Tested with two different Android devices.
- Device: Fairphone 3+
  - Nextcloud app: 3.26.0
  - OS: [/e/OS-12.3-s](https://e.foundation/) (LineageOS-19 based Android-12)
- Device: Google Pixel 4a
  - Nextcloud app: 3.26.0
  - OS: Android-13

**Comment by jospoortvliet** [comment]
Odd. This is specific to .txt files, doesn't happen in MD files?

I see you can only make 1 free line, on Text on iOS in a .txt file, but the other stuff doesn't happen. This might also be keyboard/autocomplete doing weirdo things I guess.

**Comment by kolAflash** [comment]
MD files behave buggy in a slightly different way.

<details>
<summary>video with OpenBoard and a MD file on a Fairephone 3 (e/OS/, AOSP Keyboard)</summary>

https://github.com/nextcloud/text/assets/3355089/682d8f31-045d-49e7-9c99-5cf733f94961
</details>

Same test result on the [Fairphone 3 (e/OS/)](https://github.com/nextcloud/text/issues/5103) and the [Nokia 1 (TA-1060)](https://github.com/nextcloud/text/issues/5102).

&nbsp;

Fairphone 3 (e/OS/) uses _"AOSP Keyboard"_.  
Switching to [Hacker's Keyboard](https://f-droid.org/de/packages/org.pocketworkstation.pckeyboard/) seems to workaround the issue.

[Google Pixel 4a](https://github.com/nextcloud/text/issues/5103) uses [OpenBoard](https://f-droid.org/de/packages/org.dslul.openboard.inputmethod.latin/).  
Nokia 1 uses OpenBoard too.

Using [Google GBoard](https://play.google.com/store/apps/details?id=com.google.android.inputmethod.latin) changes the behavior slightly. Newlines work with GBoard. Deleting a line (backspace) with another user cursor does only work in MD files, but not in TXT files.

&nbsp;

So the behavior with "AOSP Keyboard" and OpenBoard is broken. GBoard partly works. And only "Hacker's Keyboard" works fine.

**Comment by kolAflash** [comment]
@jospoortvliet  
Find the answers to your questions in my previous comment.

&nbsp;

I'm getting really annoyed with this.  
Since Nextcloud-26 came out [about one year ago](https://download.nextcloud.com/server/releases/) the text app is broken in many ways!  
Especially on Android!

**Unintentionally deleted text / data loss happen again and again since Nextcloud >= 26.**

&nbsp;

I invested a lot of time in exactly describing those bugs. But still the text app suffers from a lot of problems which didn't exist in Nextcloud-25.
#4100 _user cursors trigger different bugs (2023-05)_  
#5102 _Android app text editor doesn't write changes to server filesystem_  
Please feel free to ask me for anything I can provide.  
_(I'm a software developer myself and if I would have more knowledge of the Nextcloud source, I'd fix this mess myself. I just don't have enough time to properly work into the Nextcloud source so I can fix complicated synchronization issues.)_

Here's another bug video I created when my shopping list TXT file was smashed today.  
_(Android-App-3.29.0, Nextcloud-Server-28.0.5)_  
https://github.com/nextcloud/text/assets/3355089/d1b4cc7a-a900-477a-93b3-8cd16040afbb

&nbsp;

I've experienced those problems with multiple servers and on multiple end devices. So I'm very likely not the only one suffering from it. And having the constant risk of data loss _(unintentionally deleted text, unsaved files)_ is a really bad thing for the user!

**The people who introduced the changes between Nextcloud 25 and 26 _(apparently mostly adding user cursors)_ should have a look at the resulting regressions.**  
**And if no one has the resources to fix this, I suggest reverting changes which added the users cursors feature.**  
**Those user cursors are really not worth the trouble!!!**

&nbsp;

Kind regards and thanks for your work,  
kolAflash

**Comment by RoWo-DS** [comment]
We experience random text deletion and deactivated the text app for that reason. We use

- Nextcloud Hub 8 (29.0.6)
- Text 3.10.1

It would be great, if someone could fix that problem or recommend another app, which doesn't randomly delete text.

**Comment by mejo-** [comment]
Several improvements and bugfixes were added to collaborative editing in the meantime. Could you please open a new issue if you still experience issues with it?

**Comment by kolAflash** [comment]
Issue still exists as described in the initial message.

Last tested versions  
Nextcloud: `32.0.0.13`  
NC-Android-App: `3.33.0`  
Android: Android-13 ([/e/OS-3.0.4-t](https://images.ecloud.global/official/FP3/IMG-e-3.0.4-t-20250709507786-official-FP3.zip))  
Keyboard: AOSP (by /e/OS-3.0.4)

Happens in TXT files and in MD.  
For MD happens on a line inside as well as at the end of a paragraph.

**Comment by mejo-** [comment]
@kolAflash thanks for reporting this again! Can you try to reproduce this in the Tiptap example editor with collaboration cursor here: https://tiptap.dev/docs/editor/extensions/functionality/collaboration-caret

**Comment by kolAflash** [comment]
> [@kolAflash](https://github.com/kolAflash) thanks for reporting this again! Can you try to reproduce this in the Tiptap example editor with collaboration cursor here: https://tiptap.dev/docs/editor/extensions/functionality/collaboration-caret

Yes. Happens between the upper and lower [tiptap](https://github.com/ueberdosis/tiptap) demo edit field when opened in [Fennec-142.0.1](https://f-droid.org/packages/org.mozilla.fennec_fdroid/) (~Firefox) on [my  phone](https://github.com/nextcloud/text/issues/5103#issuecomment-3343349501).

**Comment by mejo-** [comment]
This prosemirror issue might be related: https://github.com/ProseMirror/prosemirror/issues/565

**Comment by mejo-** [comment]
@kolAflash we fixed a bug in Text some months back that fixed a couple of problems with inserting on Android keyboards. Could you check whether you still encounter this specific issue?

**Comment by mejo-** [comment]
Let's close this one due to lack of information.
